demo/index.html

```diff
@@ -10,6 +10,7 @@
     <div class="main-content">
         <h1>Vanilla UI—Dialog</h1>
         <button class="js-dialog-btn-open" type="button" data-controls-modal="my-dialog">Open dialog 1</button>
+        <button class="js-dialog-btn-open" type="button" data-controls-modal="my-dialog2">Open dialog 2</button>
```

Comment:
Outside of this task so I can create a GH Issue if you like? 

The data attribute should be: `data-controls-dialog` as **modal** is a type of dialog 👍. 

Also I'm thinking do we need to namespace any of our public hooks: default JS hooks and data attribute, to prevent any clashes in the consumer's document? Unlikely but good to be bulletproof. E.g.

```html
js-vui-dialog-btn-open

data-vui-controls-dialog
```